bin/tempo-bench/src/cmd/max_tps.rs

```diff
@@ -280,16 +280,20 @@ impl MaxTpsArgs {
         .await
         .context("Failed to set default fee token")?;
 
-        // Setup DEX
         let user_tokens = 2;
-        info!(user_tokens, "Setting up DEX");
-        let (quote_token, user_tokens) = dex::setup(
-            signer_providers,
-            user_tokens,
-            self.max_concurrent_requests,
-            self.max_concurrent_transactions,
-        )
-        .await?;
+        let (quote_token, user_tokens) = if self.place_order_weight > 0.0 && self.swap_weight > 0.0
+        {
+            info!(user_tokens, "Setting up DEX");
+            dex::setup(
+                signer_providers,
+                user_tokens,
+                self.max_concurrent_requests,
+                self.max_concurrent_transactions,
+            )
+            .await?
```

Comment:
Oh, this is not really correct to do, because `dex::setup` also sets up the TIP-20 tokens that we use for transfers. What we should do instead, is **always** setup the TIP-20 tokens, but the DEX setup should be done only if place order or swap weights are > 0.

Does this make sense?

Reply:
Let's proceed with option 2, we need to set up TIP-20 tokens no matter what, so it makes sense to move it to a separate function!